### ISSUE TITLE
Add fuselage structural mounts

### ISSUE THREAD

**Issue by sdeinert** [DLR-SL/CPACS#790]
Similar to the wing structural mounts a definition for structural mounts inside the fuselage is required:

For this the `fuselageStructureType` could be extended by an optional element 
```XML
<xsd:element minOccurs="0" name="fuselageStructuralMounts" type="fuselageStructuralMountsType"/>
```
with the type `fuselageStructuralMountsType`:
```XML
<xsd:complexType name="fuselageStructuralMountsType">
    <xsd:complexContent>
        <xsd:extension base="complexBaseType">
            <xsd:sequence>
                <xsd:element maxOccurs="unbounded" name="fuselageStructuralMount" type="structuralMountType"/>
            </xsd:sequence>
        </xsd:extension>
    </xsd:complexContent>
</xsd:complexType>
```
containing elements of the existing type `structuralMountType`.

This would allow defining attachments inside the fuselage analogue to the existing solution in the wing.

**Comment by jnwalther** [comment]
For those who, like me, are unaware of the details of the `structuralMountType`, [here](https://www.cpacs.de/documentation/CPACS_3_4_0_Docs/html/d88efd04-24bc-ca8d-b6ed-e971b0aeace4.htm) is a link to the docs.

To be honest I still have a hard time figuring out what exactly you would like to do based on this information. @sdeinert, could you maybe provide an image or some other kind of example for illustration?

**Comment by sdeinert** [comment]
I'll try :-)
To be able to place for example non-structural masses in your model, a definition of where these masses are attached to the structure becomes necessary. So for example you would like to model a heavy on-board system, piece of payload, fuel tank, engine, etc. in CPACS as a fixed point mass. First step would be to create a mass point in the centre of gravity of your component. For systems, engines and payload this is probably already available, for the fuel tank you would evaluate the volume and fill rate from its CPACS definition to obtain the CoG.
In the next step the mass point needs to be connected to the structure in a physically sensible way to introduce any inertia loads generated by this mass properly into the structure. And this purpose is fulfilled by the structural mounts. The logic of these mounts is to identify points or lines in the geometry created by intersecting two referenced structural components. These points or lines are then connected to the mass point centre of gravity. The reasoning behind this is that the crossings between two structural objects (e.g. rib and spar) usually represent locations in the model that are stiff enough to support the extra loads coming from any equipment.

As you can see in this picture the sphere  would be a heavy system, which is connected to the structure using **three** structural mounts
![structuralmounts](https://user-images.githubusercontent.com/15669066/184824861-ec50adf5-a847-4216-ad8f-5e02d3aa306f.png)

From left to right:
 - The left most attachment uses the end points of the intersection line between the starboard wall and the front bulkhead
 - The middle attachment uses all points (for visualization purpose only 7 points are drawn here) on the intersection line between center wall and front bulkhead
 - The right most attachment has a similar definition to the left one using end points of the intersection line between port wall and front bulkhead.

When creating a structural FEM model of this CPACS representation, a rigid body element (e.g. Nastran RBEx) can be generated based on the yellow lines.
This is for example also used to attach landing gear struts to the aircraft structure.

In short: structural mounts identify points or lines by intersecting two structural components, these resulting points or lines are then connected to a third component which references this structural mount as its attachment. Usually, as in the example above, components reference more than one structural mounts as their attachment points.

The structural mounts represent our current solution of attaching things to the aircraft structure after they have been placed in the aircraft. In case there are any better solutions or ideas for this problem, I am happy to discuss them here.

**Comment by jnwalther** [comment]
Thanks for the explanation! The image helps. I'm wondering, does the strucutral mount have any kind of physical counterpart or is it rather an easy way to formulate RBEs in CPACS? If the latter is the case, I have to say, I'm not a fan. As a structural engineer, I see the appeal, but I feel it dilutes, what the structure node stands for, i.e. the _aircraft structure_. Not the aircraft structure _FEM model_! That said, I guess I'm a little late now, since the wing counterpart is already in CPACS. I also can't come up with a better solution right now, so I guess the above qualifies as food for thought...

**Comment by sdeinert** [comment]
As the placement and attachment often drives the structural design, I would say that the definition of attachments is very much a part of the aircraft structure. Without such knowledge you wouldn't know where to place your structure or where to reinforce it, since you have to expect high inertia loads in certain areas.

I am happy to discuss better solutions, but without a definition fulfilling the purpose of the structural mounts the link between system design/layout and structural design is missing.

**Comment by MarAlder** [comment]
I added the description above to the [documentation](http://dlr-sl.github.io/CPACS/html/d88efd04-24bc-ca8d-b6ed-e971b0aeace4.htm).
